packages/composites/token-allocation/src/config.ts

```diff
@@ -1,20 +1,42 @@
 import { util } from '@chainlink/ea-bootstrap'
 import { getDefaultConfig } from '@chainlink/external-adapter'
 import { getDataProvider } from './dataProvider'
-import { Config } from './types'
+import { Config, SourceRequestOptions } from './types'
+import { adapters } from './dataProvider'
 
 export const DEFAULT_TOKEN_DECIMALS = 18
 export const DEFAULT_TOKEN_BALANCE = 1
 
+export const ENV_DATA_PROVIDER_URL = 'DATA_PROVIDER_URL'
+
+export const getURL = (prefix: string, required = false) =>
+  required
+    ? util.getRequiredEnv(ENV_DATA_PROVIDER_URL, prefix)
+    : util.getEnv(ENV_DATA_PROVIDER_URL, prefix)
+
 export const makeConfig = (prefix = ''): Config => {
-  const dataProviderUrl = util.getRequiredEnv('DATA_PROVIDER_URL', prefix)
-  const defaultConfig = getDefaultConfig(prefix)
-  defaultConfig.api.baseURL = defaultConfig.api.baseURL || dataProviderUrl
-  defaultConfig.api.method = 'post'
+  const sources: SourceRequestOptions = {}
+
+  for (const a of adapters) {
+    const url = getURL(a.NAME.toUpperCase())
+    if (url) {
+      const defaultConfig = getDefaultConfig(prefix)
+      defaultConfig.api.baseURL = url
+      defaultConfig.api.method = 'post'
+      sources[a.NAME] = getDataProvider(defaultConfig.api)
```

Comment:
To avoid case errors, make the sources keys either uppercase or lowercase, and make the conversion as well when you want to access any of them